llvm/lib/IR/ConstantFold.cpp

```diff
@@ -1154,6 +1154,9 @@ static ICmpInst::Predicate evaluateICmpRelation(Constant *V1, Constant *V2) {
                                 GV->getType()->getAddressSpace()))
         return ICmpInst::ICMP_UGT;
     }
+  } else if (const ConstantPtrAuth *SP = dyn_cast<ConstantPtrAuth>(V1)) {
+    // FIXME: ahmedbougacha: implement ptrauth cst comparison
+    return ICmpInst::BAD_ICMP_PREDICATE;
```

Comment:
Looking at this, I noticed that the existing code here is broken, fixed in https://github.com/llvm/llvm-project/commit/00ca80938b732ee43deb2a94bec1c7efef3025d4. With that, you probably don't need this placeholder to prevent crashes anymore.

Reply:
Oh, nice, thanks;  this deserves to be covered regardless, and the previous (non-Constant) iteration had some logic here, but let's do that separately.